components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt

```diff
@@ -244,6 +244,11 @@ abstract class EngineSession(
      */
     abstract fun exitFullScreenMode()
 
+    /**
+     * Tries to recover from a crash by restoring the last know state.
+     */
+    abstract fun recoverFromCrash()
```

Comment:
Question: As the comment implies  **Tries to recover**, that means that in some cases it won't be able to recover from a crash, should we indicate to the consumer when we are unable to recover like returning a Boolean? This way the consumer can handle the situation properly.

Reply:
Or  `session.crashed` is the way to communicate it ?

Reply:
We could return a `Boolean` indicating whether we had a state and whether we tried to restore that. I'll add that.

Reply:
👍